apps/client/src/app/modules/list/data/ListRowSerializationHelper.ts

```diff
@@ -0,0 +1,315 @@
+  }
+  private getJobAbbreviationFromId(job: number): string {
+    const jobAbbr = this.getNameIfExists(this.l12n.getJobAbbr(job));
+    return jobAbbr != null && jobAbbr !== "missing name" ? jobAbbr : "DOWM";
+  }
+
+  private getNameIfExists(name: I18nName): string {
+    const translatedName = this.i18nTools.getName(name);
+    return translatedName && translatedName !== 'no name' ? translatedName : undefined;
+  }
+
+  private serializeVoyages(voyages: any): string {
+    return voyages && voyages.length > 0 ? voyages.map((r: any) => this.getNameIfExists(r)) : undefined;
+  }
+
+  public applyNpcName(obj: Vendor): any {
+    return {
+      ...obj,
+      npcName: this.getNameIfExists(this.l12n.getNpc(obj.npcId)),
+      zoneName: this.getZoneName(obj),
+    };
+  }
+
+  private getZoneName(obj: MapInfo): string {
+    if (!obj)
+      return undefined;
+    ///wat.. casing!?
+    const zoneName = this.l12n.getPlace(obj.zoneId ? obj.zoneId : obj.zoneid);
+    const mapName = this.l12n.getMapName(obj.mapId);
+    return this.getNameIfExists(zoneName ? zoneName : mapName);
+  }
+
+  public serializeTrades(trades: NpcTradeData[]): NpcTradeData[] {
+    return trades && trades.length > 0 ? trades.map((r: any) => this.serializeTradeData(r)) : undefined;
+  }
+
+  public serializeTrade(obj: CurrencyInfo): CurrencyInfo {
+    return {
+      currencies: obj.currencies && obj.currencies.length > 0 ? obj.currencies.map(c => this.applyItemName(c)) : undefined,
+      items: obj.items && obj.items.length > 0 ? obj.items.map(i => this.applyItemName(i)) : undefined,
+    };
+  }
+
+  public serializeTradeData(obj: NpcTradeData): NpcTradeData {
+    return {
+      ...obj,
+      npcs: this.serializeNPCs(obj.npcs),
+      trades: obj.trades ? obj.trades.map(t => this.serializeTrade(t)) : undefined,
+    };
+  }
+
+  public applyMonsterInfo(obj: MonsterInfo): MonsterInfoWithMobName {
+    const id = obj.id ? obj.id : obj.itemId ? obj.itemId : undefined;
+    const mobName = this.getMobNameFromId(id);
+    const retval = {
+      ...obj,
+      name: this.getNameIfExists(mobName),
+      zoneName: this.getZoneName(obj.position ? obj.position : obj),
+    };
+    return retval;
+  }
+
+  private getMobNameFromId(id: number): I18nName {
+    return this.l12n.getMob(MobNamePipe.getActualMobId(id));
+  }
+
+  private getMonsterHuntData(monsterDrops: MonsterInfo[]): MonsterInfoWithMobName[] {
+    return monsterDrops && monsterDrops.length > 0 ? monsterDrops.map((r: any) => this.applyMonsterInfo(r)) : undefined;
+  }
+
+  public serializeNPCs(vendors: Vendor[]): VendorWithName[] {
+    return vendors && vendors.length > 0 ? vendors.map((r: Vendor) => this.applyNpcName(r)) : undefined;
+  }
+
+  private serializeVentures(item: ListRow, ventures: number[]) {
+    return ventures && ventures.length > 0 ? this.gt.getVentures(ventures).map(venture => {
+      const retval = {
+        ...venture,
+        ventureDetails: VenturesComponent.ventureAmounts(venture)
```

Comment:
Calling a static method from a component here is not acceptable from a design pattern point of view. Angular provides dependency injection and Typescript provides inheritence so you can avoid referencing static code accross multiple classes, making dependncies way easier to understand and organize.